File: golang/015.go
package main

import "fmt"

func findLargestSum(arr []int) int {
	maxSum := arr[0]
	sum := arr[0]

	for i:=1; i < len(arr);i++{
		if sum + arr[i] < arr[i]{
			sum = arr[i]
		} else {
			sum = sum + arr[i]
		}
		if maxSum < sum{
			maxSum = sum
		}
	}

	return maxSum
}

func main() {
	fmt.Println(findLargestSum([]int{5,4,-1,7,8}))
}

File: python/015.py
def findLargestSum(arr):
    sum = arr[0]
    maxSum = arr[0]

    for x in range(1,len(arr)):
        # sets the value of sum depending on value of current element
        sum = max(sum + arr[x], arr[x])
        # sets value of maxSum btwn sum and maxSum
        maxSum = max(maxSum, sum)
    
    return maxSum

print(findLargestSum([-2,1,-3,4,-1,2,1,-5,4]))